[apex] java.lang.NoClassDefFoundError in the latest build

**Rule Set:**
apex-security, vf-security
**Description:**
Apex parser broken in the latest build. Stack trace is below:
```
$ ./run.sh pmd -d ~/myApexPackage -f text -rulesets apex-security
Exception in thread "main" java.lang.NoClassDefFoundError: shaded/org/objectweb/asm/Handle
	at apex.jorje.semantic.ast.context.Emitter.<clinit>(Emitter.java:42)
	at apex.jorje.semantic.compiler.CompilerContext.<init>(CompilerContext.java:50)
	at apex.jorje.semantic.compiler.ApexCompiler.<init>(ApexCompiler.java:57)
	at apex.jorje.semantic.compiler.ApexCompiler.<init>(ApexCompiler.java:37)
	at apex.jorje.semantic.compiler.ApexCompiler$Builder.build(ApexCompiler.java:224)
	at net.sourceforge.pmd.lang.apex.ast.CompilerService.compile(CompilerService.java:92)
	at net.sourceforge.pmd.lang.apex.ast.CompilerService.visitAstsFromStrings(CompilerService.java:87)
	at net.sourceforge.pmd.lang.apex.ast.CompilerService.visitAstFromString(CompilerService.java:75)
	at net.sourceforge.pmd.lang.apex.ast.ApexParser.parseApex(ApexParser.java:44)
	at net.sourceforge.pmd.lang.apex.ast.ApexParser.parse(ApexParser.java:53)
	at net.sourceforge.pmd.lang.apex.ApexParser.parse(ApexParser.java:37)
	at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:113)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:175)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:97)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:52)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:93)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:27)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: shaded.org.objectweb.asm.Handle
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 23 more
```
**Running PMD through:** *CLI*

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -233,6 +233,7 @@ a warning will now be produced suggesting users to adopt it for better performan
     *   [#488](https://github.com/pmd/pmd/pull/488): \[apex] Use Apex lexer for CPD
     *   [#489](https://github.com/pmd/pmd/pull/489): \[apex] Update Apex compiler
     *   [#500](https://github.com/pmd/pmd/issues/500): \[apex] Running through CLI shows jorje optimization messages
+    *   [#605](https://github.com/pmd/pmd/issues/605): \[apex] java.lang.NoClassDefFoundError in the latest build
     *   [#637](https://github.com/pmd/pmd/issues/637): \[apex] Avoid SOSL in loops
 *   cpp
     *   [#448](https://github.com/pmd/pmd/issues/448): \[cpp] Write custom CharStream to handle continuation characters

--- a/pmd-apex/pom.xml
+++ b/pmd-apex/pom.xml
@@ -59,43 +59,6 @@
           </execution>
         </executions>
       </plugin>
-
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-shade-plugin</artifactId>
-        <executions>
-          <execution>
-            <phase>package</phase>
-            <goals>
-              <goal>shade</goal>
-            </goals>
-            <configuration>
-              <artifactSet>
-                <includes>
-                  <include>apex</include>
-                </includes>
-                <excludes>
-                  <exclude>${project.groupId}</exclude>
-                </excludes>
-              </artifactSet>
-              <relocations>
-                <!-- apex uses an older version of asm, hence we need to shaded it away -->
-                <relocation>
-                  <pattern>org.objectweb.asm</pattern>
-                  <shadedPattern>shaded.org.objectweb.asm</shadedPattern>
-                </relocation>
-                <!-- apex uses an older version of jackson, hence we need to shaded it away -->
-                <relocation>
-                  <pattern>com.fasterxml.jackson</pattern>
-                  <shadedPattern>shaded.com.fasterxml.jackson</shadedPattern>
-                </relocation>
-              </relocations>
-              <shadedArtifactAttached>true</shadedArtifactAttached>
-              <shadedClassifierName>apex-jorje-shaded</shadedClassifierName>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
     </plugins>
   </build>
   <dependencies>
@@ -165,6 +128,12 @@
     <dependency>
       <groupId>net.sourceforge.saxon</groupId>
       <artifactId>saxon</artifactId>
+    </dependency>
+
+    <dependency>
+        <groupId>org.ow2.asm</groupId>
+        <artifactId>asm</artifactId>
+        <scope>runtime</scope>
     </dependency>
 
     <dependency>

--- a/pmd-dist/pom.xml
+++ b/pmd-dist/pom.xml
@@ -209,12 +209,6 @@
                 </dependency>
                 <dependency>
                     <groupId>net.sourceforge.pmd</groupId>
-                    <artifactId>pmd-apex</artifactId>
-                    <version>${project.version}</version>
-                    <classifier>apex-jorje-shaded</classifier>
-                </dependency>
-                <dependency>
-                    <groupId>net.sourceforge.pmd</groupId>
                     <artifactId>pmd-ui</artifactId>
                     <version>${project.version}</version>
                 </dependency>

--- a/pmd-dist/src/main/assembly/bin.xml
+++ b/pmd-dist/src/main/assembly/bin.xml
@@ -52,10 +52,6 @@
             <directoryMode>0755</directoryMode>
             <fileMode>0644</fileMode>
             <useProjectArtifact>false</useProjectArtifact>
-            <!-- exclude the apex (transitive) dependencies - we use the shaded version instead -->
-            <excludes>
-                <exclude>apex:*</exclude>
-            </excludes>
         </dependencySet>
     </dependencySets>
 </assembly>

